khelo/lib/ui/flow/tournament/add/add_tournament_screen.dart

```diff
@@ -0,0 +1,275 @@
+
+import '../../../../components/action_bottom_sheet.dart';
+import '../../../../components/app_page.dart';
+import '../../../../components/error_snackbar.dart';
+import '../../../../components/image_picker_sheet.dart';
+import '../../../../domain/formatter/date_formatter.dart';
+
+class AddTournamentScreen extends ConsumerStatefulWidget {
+  const AddTournamentScreen({super.key});
+
+  @override
+  ConsumerState<AddTournamentScreen> createState() =>
+      _AddTournamentScreenState();
+}
+
+class _AddTournamentScreenState extends ConsumerState<AddTournamentScreen> {
+  late AddTournamentViewNotifier notifier;
+
+  @override
+  void initState() {
+    super.initState();
+    notifier = ref.read(addTournamentStateProvider.notifier);
+  }
+
+  void _observeActionError(BuildContext context, WidgetRef ref) {
+    ref.listen(addTournamentStateProvider.select((value) => value.actionError),
+        (previous, next) {
+      if (next != null) {
+        showErrorSnackBar(context: context, error: next);
+      }
+    });
+  }
+
+  void _observePop(BuildContext context, WidgetRef ref) {
+    ref.listen(addTournamentStateProvider.select((value) => value.pop),
+        (previous, next) {
+      if (next) {
+        context.pop();
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    _observeActionError(context, ref);
+    _observePop(context, ref);
+
+    final state = ref.watch(addTournamentStateProvider);
+
+    return AppPage(
+      title: context.l10n.add_tournament_screen_title,
+      body: Builder(
+        builder: (context) => Stack(
+          children: [
+            _body(context, state),
+            _stickyButton(context, state),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Widget _body(BuildContext context, AddTournamentState state) {
+    return ListView(
+      padding: context.mediaQueryPadding + const EdgeInsets.all(16),
+      children: [
+        _profileView(context, state),
+        AppTextField(
+          controller: state.nameController,
+          label: context.l10n.add_tournament_name,
+          onChanged: (_) => notifier.onChange(),
+          contentPadding:
+              const EdgeInsets.symmetric(horizontal: 16, vertical: 12),
+          borderRadius: BorderRadius.circular(12),
+          borderType: AppTextFieldBorderType.outline,
+          borderColor: BorderColor(
+            focusColor: context.colorScheme.outline,
+            unFocusColor: context.colorScheme.outline,
+          ),
+        ),
+        const SizedBox(height: 16),
+        _selectTileView(
+          label: context.l10n.add_tournament_type,
+          title: state.selectedType.getString(context),
+          icon: Icons.keyboard_arrow_down_rounded,
+          onTap: () {
+            _selectTypeSheet(
+              context,
+              selectedType: state.selectedType,
+              onSelected: (type) => notifier.onSelectType(type),
+            );
+          },
+        ),
```

Comment:
check out **AdaptiveOutlinedTile** available in our Widgets, we can reuse it here.